Sources/SwiftFusion/Core/Tuple+EuclideanVectorN.swift

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import PenguinStructures
+
+extension Empty: AdditiveArithmetic {
+  public static func + (_ lhs: Self, _ rhs: Self) -> Self {
+    return Empty()
+  }
+  public static func - (_ lhs: Self, _ rhs: Self) -> Self {
+    return Empty()
+  }
+  public static var zero: Self {
+    return Empty()
+  }
+}
+
+extension Tuple: AdditiveArithmetic where Head: AdditiveArithmetic, Tail: AdditiveArithmetic {
```

Comment:
Not a definitive thing whatsoever, just a note:  I'm a little wary about conditionally conforming `Tuple` to every possible thing for the same reasons I don't think `Array<Layer>` should conform to `Layer`.  It's kind of sad that we don't have scoped conformances, because that would make this much less of an issue.